modules/configuration/src/main/java/org/apache/ignite/internal/configuration/ConfigurationRegistry.java

```diff
@@ -137,16 +131,10 @@ public void stop() throws Exception {
     }
 
     /**
-     * Initializes the configuration storage - reads data and sets default values for missing configuration properties.
+     * Persist default values to the storage.
      */
-    public void initializeDefaults() {
-        changer.initializeDefaults();
-
-        for (RootKey<?, ?> rootKey : rootKeys) {
-            DynamicConfiguration<?, ?> dynCfg = configs.get(rootKey.key());
-
-            touch(dynCfg);
-        }
```

Comment:
Why have you removed this code? I believe it may be necessary for notifications to work properly, but I have no tests to support the statement. Did you fully understand the intention of the code before removing it?

Reply:
It was used in production before. This is interesting, I guess our tests are insufficient, this sucks. I clearly remember, that there was a reason for this code. Let's discuss it right after current issue and maybe try writing a test that would fail without this code.